crates/red_knot_python_semantic/src/types/display.rs

```diff
@@ -20,41 +22,64 @@ pub struct DisplayType<'db> {
+            Type::None => "None".to_string(),
             Type::Module(file) => {
-                write!(f, "<module '{:?}'>", file.path(self.db))
+                format!("<module '{:?}'>", file.path(self.db))
             }
             // TODO functions and classes should display using a fully qualified name
-            Type::Class(class) => write!(f, "Literal[{}]", class.name(self.db)),
-            Type::Instance(class) => f.write_str(&class.name(self.db)),
-            Type::Function(function) => write!(f, "Literal[{}]", function.name(self.db)),
-            Type::Union(union) => union.display(self.db).fmt(f),
-            Type::Intersection(intersection) => intersection.display(self.db).fmt(f),
-            Type::IntLiteral(n) => write!(f, "Literal[{n}]"),
+            Type::Class(class) => class.name(self.db).to_string(),
+            Type::Instance(class) => class.name(self.db).to_string(),
+            Type::Function(function) => function.name(self.db).to_string(),
+            Type::Union(union) => union.display(self.db).to_string(),
+            Type::Intersection(intersection) => intersection.display(self.db).to_string(),
+            Type::IntLiteral(n) => n.to_string(),
             Type::BooleanLiteral(boolean) => {
-                write!(f, "Literal[{}]", if *boolean { "True" } else { "False" })
+                if *boolean {
+                    "True".to_string()
+                } else {
+                    "False".to_string()
+                }
+            }
+            Type::StringLiteral(string) => {
+                format!(r#""{}""#, string.value(self.db).replace('"', r#"\""#))
             }
-            Type::StringLiteral(string) => write!(
-                f,
-                r#"Literal["{}"]"#,
-                string.value(self.db).replace('"', r#"\""#)
-            ),
-            Type::LiteralString => write!(f, "LiteralString"),
+            Type::LiteralString => "LiteralString".to_string(),
             Type::BytesLiteral(bytes) => {
                 let escape =
                     AsciiEscape::with_preferred_quote(bytes.value(self.db).as_ref(), Quote::Double);
+                escape.bytes_repr().to_string().unwrap_or_default()
+            }
+        }
+    }
+}
 
+impl Display for DisplayType<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self.ty {
+            // Literal types -> `Literal[<repr>]`
+            Type::Class(_)
+            | Type::Function(_)
+            | Type::IntLiteral(_)
+            | Type::BooleanLiteral(_)
+            | Type::StringLiteral(_)
+            | Type::BytesLiteral(_) => {
                 f.write_str("Literal[")?;
-                escape.bytes_repr().write(f)?;
+                f.write_str(&self.representation())?;
                 f.write_str("]")
             }
+            // Non literal types -> `<repr>`
+            _ => f.write_str(&self.representation()),
         }
```

Comment:
I think I would simplify this code by:

1. Adding a `is_literal` method to `Type` that returns true if `self` is a `Class`, `Function`, Int`, Boolean`, `String` or `Bytes` Literal. 
2. Change the method here to:

```rust
let is_literal = self.ty.is_literal();

if is_literal {
	f.write_str("Literal[")?;
}

match self.ty {
	...
}

if is_literal {
	f.write_str("]")?;
}

```

Reply:
I thought about adding this information to `Type` as it's more elegant, but I wasn't confident enough to code outside of `display.rs` :)
Thank you!

Reply:
I thought about having this information in `Type` as it's way cleaner, but I was hesitant to code outside of `display.rs`. Thank you for the suggestion 🙏